File: 2405.1/phpchk/security/MissingLoginControl/MissingLoginControl.php
<?php

function login() {
  for($i=10; $i<10; $i++) {
    // Login here ...
  }
}

function anotherLogin() {
  $i = 0;
  while($i<10) {
    // Login here ...
  }
}

function another2Login() {
  $i = 0;
  do {
    // Login here ...
  } while($i<10);
}

function another3Login() {
  for(;;) { // @violation
    // Login here ...
  }
}

function another4Login() {
  $i = 0;
  while(((true))) { // @violation
    // Login here ...
  }
}

function f1() {
  login(); // @violation
}

function f2() {
  for($i=0; $i<10; $i++) {
    login();
  }
}

function f3() {
  $i = 0;
  while($i < 10) {
    login();
    $i++;
  }
}

function f4() {
  for($i=0; true; $i++) {
    login(); // @violation
  }
}

function f5() {
  $i = 0;
  while(((true))) {
    login(); // @violation
    $i++;
  }
}

function f6() {
  do {
    login(); // @violation
  } while(true);
}

File: 2405.1/jf_syn/security/MissingLoginControlChecker/MissingLoginControl.java
package security;
import testcasesupport.KrdUtil;
public class MissingLoginControl {

    private void login() {
        for(int i=0; i<10; i++) {
            // Log-in here ...
        }
    }

    private void anotherlogin() {
        int i;
        while(i < 10) {
            // Log-in here ...
        }
    }

    private void another2login() {
        int i;
        do {
            // Log-in here ...
        } while(i < 10);
    }

    private void another3login() {
        for(;;) { // @violation
            // Log-in here ...
        }
    }

    private void another4login() {
        int i;
        while(((true))) { // @violation
            // Log-in here ...
            i++;
        }
    }

    public void func1() {
        login(); // @violation
    }

    public void func2() {
        for(int i=0; i<10; i++) {
            login();
        }
    }

    public void func3() {
        int i;
        while(i < 10) {
            login();
            i++;
        }
    }

    public void func4() {
        for(int i=0; true; i++) {
            login(); // @violation
        }
    }

    public void func5() {
        int i;
        while(((true))) {
            login(); // @violation
            i++;
        }
    }

    public void func6() {
        do {
            login(); // @violation
        } while(true);
    }

    public void bad(HttpServletRequest request, HttpServletResponse response) throws Throwable
    {
        String username = request.getParameter("username");
        String password = request.getParameter("password");

        /* FLAW */
        if (username == null || password == null || !KrdUtil.isAuthenticatedUser(username, password)) // @violation
        {
            response.getWriter().println("Invalid username, password<br/>");
            response.getWriter().println("<A href='/KRD_Debugging/KRD_214_Improper_Restriction_Of_Excessive_Authentication_Attempts.htm'>Login Again</A></body></html>");
        }
        else
        {
            response.getWriter().println(username + " has successfully logged in!");
        }
    }
}
